lib/oxidized/model/riverbed.rb

```diff
@@ -0,0 +1,131 @@
+    comments = []
+    cfg.each_line do |line|
+      line.strip!
+      comments << "Product name: #{Regexp.last_match(1)}" if line =~ /^Product name:\s+(.*)$/
+      comments << "Product release: #{Regexp.last_match(1)}" if line =~ /^Product release:\s+(.*)$/
+      comments << "Build ID: #{Regexp.last_match(1)}" if line =~ /^Build ID:\s+(.*)$/
+      comments << "Build date: #{Regexp.last_match(1)}" if line =~ /^Build date:\s+(.*)$/
+      comments << "Build arch: #{Regexp.last_match(1)}" if line =~ /^Build arch:\s+(.*)$/
+      comments << "Built by: #{Regexp.last_match(1)}" if line =~ /^Built by:\s+(.*)$/
+      comments << "Product model: #{Regexp.last_match(1)}" if line =~ /^Product model:\s+(.*)$/
+      comments << "Number of CPUs: #{Regexp.last_match(1)}" if line =~ /^Number of CPUs:\s+(.*)$/
+    end
+    @version_info = comments.join("\n")
+    ''
+  end
+
+  # Get hardware information and store it
+  cmd 'show hardware all' do |cfg|
+    # Remove the command echo and the hostname from the output
+    cfg = cfg.cut_head
+    cfg.gsub!(PROMPT, '')
+
+    # Extract relevant information
+    comments = []
+    cfg.each_line do |line|
+      line.strip!
+      comments << "Hardware revision: #{Regexp.last_match(1)}" if line =~ /^Hardware revision:\s+(.*)$/
+      comments << "Mainboard: #{Regexp.last_match(1)}" if line =~ /^Mainboard:\s+(.*)$/
+      if line =~ /^Slot (\d+):\s+\.*\s+(.*)$/
+        slot_number = Regexp.last_match(1)
+        slot_info = Regexp.last_match(2)
+        comments << "Slot #{slot_number}: #{slot_info}"
+      end
+      comments << "System led: #{Regexp.last_match(1)}" if line =~ /^System led:\s+(.*)$/
+    end
+    @hardware_info = comments.join("\n")
+    ''
+  end
+
+  # Get serial information from 'show info' command
+  cmd 'show info' do |cfg|
+    # Remove the command echo and the hostname from the output
+    cfg = cfg.cut_head
+    cfg.gsub!(PROMPT, '')
+
+    # Extract the 'Serial:' line
+    cfg.each_line do |line|
+      line.strip!
+      @serial_info = "Serial: #{Regexp.last_match(1)}" if line =~ /^Serial:\s+(.*)$/
+    end
+    ''
+  end
+
+  # Get the running configuration
+  cmd 'show running-config' do |cfg|
+    # Remove the first and last lines (command echo and hostname)
+    cfg = cfg.cut_head.cut_tail
+
+    # Process lines containing '##'
+    cfg = cfg.each_line.map do |line|
+      if line =~ /^(.*##.*?##)(.*)$/
+        # Split the line into comment and rest
+        comment_part = Regexp.last_match(1).strip
+        command_part = Regexp.last_match(2).strip
+        comment_line = comment(comment_part)
+        if command_part.empty?
+          comment_line + "\n"
+        else
+          comment_line + "\n" + command_part + "\n"
+        end
+      else
+        line
+      end
+    end.join
+
+    # Prepend date, version, hardware, and serial information to the configuration
+    header = []
+    header << comment("Date of version: #{Time.now.strftime('%Y-%m-%d %H:%M:%S %Z')}")
```

Comment:
Printing the time is not a good idea, as oxidized will store a config after every fetch. The idea is to strip everything changing out, and store the configs in git, so you only get differences and not thousand of identical files.

Reply:
Removed the line that adds the date to the header.